action.py

```diff
@@ -818,13 +818,22 @@ def import_annotations(self, reader_app):
             reader_app = exporting_apps[reader_app_class]
 
             # Open the Import Annotations dialog
-            raw_xml = ImportAnnotationsDialog(self, reader_app, reader_app_class).text
-            if(raw_xml):
+            if reader_app_class.SUPPORTS_FILE_CHOOSER:
+                raw_data = ImportAnnotationsFileDialog(self, reader_app_class).text()
+            else:
+                raw_data = ImportAnnotationsTextDialog(self, reader_app, reader_app_class).text()
+            if(raw_data):
                 # Instantiate reader_app_class
                 rac = reader_app_class(self)
-                success = rac.parse_exported_highlights(raw_xml)
+                success = rac.parse_exported_highlights(raw_data)
                 if not success:
-                    self._log("errors parsing raw_xml")
+                    self._log("errors parsing data for import")
+                    msg = QMessageBox()
+                    msg.setIcon(QMessageBox.Critical)
+                    msg.setText("Import Error")
+                    msg.setInformativeText('Error parsing data.')
+                    msg.setWindowTitle("Import Error")
+                    msg.exec_()
```

Comment:
So now, when parsing fails user will see a nice error window? Before this change user wasn't informed about a failure, right?

Reply:
Yea, it failed silently before.

Reply:
Can you please make a screenshot of that?

Reply:
I don't think I'll be doing anything else on these PR's until @davidfor acknowledges them or my posts on mobile read. I don't mean to be impatient, but I also don't want to invest time in something that nobody cares about.

Reply:
That's fair.